polyply/src/presistence.py

```diff
@@ -0,0 +1,143 @@
+    L and presistence length _lambda according to the Worm-Like-Chain
+    model. The equation was published in: "Lee, H. and Pastor, R.W., 2011.
+    Coarse-grained model for PEGylated lipids: effect of PEGylation on the
+    size and shape of self-assembled structures. The Journal of Physical
+    Chemistry B, 115(24), pp.7830-7837."
+
+    Parameters
+    ----------
+    h: float
+        the end-to-end distance
+    L: float
+        the contour length
+    _lambda: float
+        the presistence length
+
+    Returns
+    -------
+    float
+        probability to find the end-to-end distance
+    """
+    alpha = 3*L/(4*_lambda)
+    C = (np.pi**3/2. * np.exp(-alpha)*alpha**(-3/2.)*(1 + 3/(alpha) + (15/4)/alpha**2.))**-1. 
+    A = 4*np.pi*h**2.*C
+    B = L*(1-(h/L)**2.)**9/2.
+    D = -3*L
+    E = 4*_lambda*(1-(h/L)**2.)
+    return A/B * np.exp(D/E)
+
+
+DISTRIBUTIONS = {"WCM": worm_like_chain_model, }
+
+def generate_end_end_distances(molecule, specs, nonbond_matrix, seed=None):
+    """
+    Subsample a distribution of end-to-end distances given a
+    presistence length, residue graph, and theoretical model.
+
+    Parameters
+    ----------
+    molecule: `class:nx.Graph`
+    specs: `tuple`
+        named tuple with attributes model, lp (i.e. presistence length),
+        start, stop node indices, which define the ends and mol_idxs
+        which are the indices of the batch of molecules.
+    nonbond_matrix: `:class:polyply.src.nb_engine.NonBondMatrix`
+    seed: int
+        random seed for subsampling distribution
+
+    Returns
+    -------
+    np.ndarray
+        an array of end-to-end distances with shape (1, len(mol_idxs))
+    """
+    # compute the shortest path between the ends in graph space
+    end_to_end_path = nx.algorithms.shortest_path(molecule,
+                                                  source=specs.start,
+                                                  target=specs.stop)
+    end_to_end_path = list(zip(end_to_end_path[:-1], end_to_end_path[1:]))
+    # compute the length of that path in cartesian space
+    mol_idx = specs.mol_idxs[0]
+    max_path_length = _compute_path_length_cartesian(molecule,
+                                                     mol_idx,
+                                                     end_to_end_path,
+                                                     nonbond_matrix)
+    # define range of end-to-end distances
+    # increment is the average step length
+    incrm = max_path_length / len(end_to_end_path)
+    ee_distances = np.arange(incrm, max_path_length, incrm)
+    np.random.seed(seed)
+    # generate the distribution and sample it
+    probabilities = DISTRIBUTIONS[specs.model](ee_distances, max_path_length, specs.lp)
+    #TODO
+    #replace choice by https://numpy.org/doc/stable/reference/random/generated/
+    #                          numpy.random.Generator.choice.html#numpy.random.Generator.choice
+    np.random.seed(seed)
```

Comment:
Why seed twice?
Also, regarding the seed, the new paradigm is to provide a seeded generator (I think). random.seed might be deprecated

Reply:
no reason just an accident. Yes I'm aware hence the TODO the line above. But also at the moment the seed generator gives issues with other functions/libraries (see nx issue #4718 for example). I need to do proper follow up and then switch all random seeding to this paradigm. But I want to make that subject to a different PR